### ISSUE TITLE
FitBUM distribution error

### ISSUE THREAD

**Issue by deevdevil88** [floklimm/scPPIN#3]
Hello,
I have been trying to run FitBum function on the Pvalue vector for a particular cell. It keeps saying optimization failed. In the FAQs there is a solution in regards to pvalues not being in the 0-1 interval, but i ran the checksum test suggested in tutorial 1 and that returned a value of 0L.

So what might be the reason for this.

Best,
Devika

**Comment by floklimm** [comment]
That the checksum is zero means that all your p-values are in (0,1]. That your optimization fails indicates that the p-values don't follow the beta-uniform model (BUM).

Maybe inspect your p-value distribution in a histogram and check whether you see any unexpected behavior. 

If you don't see anything obviously wrong you can always create node-weights from p-values in a less statistically rigorous (but still reasonable) way:

node score = log(p-value) + X 
where X in (-Inf,+Inf) is an offset, which varies the size of the functional module you will get. The smaller the offset, the smaller the module you obtain. 

A good starting point to play around with X is around -1. (At least in some experiments we did for our data.)

**Comment by deevdevil88** [comment]
Thanks for getting back. I think you are right there is something wrong with the pvalue distribution. As i was using the multiple test corrected p-values, and i guess for some cell types the distribution was more  heavily inclined to very low p-values than 1s and perhaps thats why the pvalues dont follow the BUM . When i reran the process for these cell types using the uncorrected pvalues it ran fine.  I was wondering, in the paper and the tutorial are you using the corrected or un-corrected pvalues?
I will rerun the analysis by playing around with the X.

Thanks,
Devika

**Comment by floklimm** [comment]
It is best if you use uncorrected p-values to ensure that they are all in (0,1]. That's what we did in the paper, too.

Glad to hear that it works with the uncorrected p-values.

**Comment by deevdevil88** [comment]
Thanks Florian. 
another question did you always get a high (> 0.05) p-value so that you could accept the null hypothesis of the KS test that the distributions are similar. Because i dont always get a p> 0.05 ,but get a low D statistic.

Example using un-corrected pvalues

	One-sample Kolmogorov-Smirnov test

data:  AT_DAT_H_pValues
D = 0.030435, p-value = 2.15e-09
alternative hypothesis: two-sided

![image](https://user-images.githubusercontent.com/33758226/70623752-ad441580-1c16-11ea-9bd4-a1bb6e023889.png)

**Comment by floklimm** [comment]
In the example we discuss in the SI of our paper we do:
"The BUM is in decent agreement with the observed p-values (a one-sample Kolmogorov–Smirnov test yields a statistic of D ≈ 0.014 and a p-value of 0.086)."

We got similar results for other data we analysed.

To be honest, I wouldn't worry too much if the KS-test is not significant: you still can use the scPPIN pipeline and the scoring function. But you probably shouldn't completely trust the false discovery rates that we specify.
